geode-redis/src/main/java/org/apache/geode/redis/internal/GeodeRedisServer.java

```diff
@@ -462,37 +445,25 @@ private void initializeRedis() {
         hLLRegion = regionFactory.create(HLL_REGION);
       }
 
-      if ((redisHash = cache.getRegion(HASH_REGION)) == null) {
-        RegionFactory<ByteArrayWrapper, RedisHash> regionFactory =
-            gemFireCache.createRegionFactory(DEFAULT_REGION_TYPE);
-        redisHash = regionFactory.create(HASH_REGION);
-      }
-
-      if ((redisSet = cache.getRegion(SET_REGION)) == null) {
-        RegionFactory<ByteArrayWrapper, RedisSet> regionFactory =
-            gemFireCache.createRegionFactory(DEFAULT_REGION_TYPE);
-        redisSet = regionFactory.create(SET_REGION);
-      }
-
-      if ((redisMetaData = cache.getRegion(REDIS_META_DATA_REGION)) == null) {
-        InternalRegionFactory<String, RedisDataType> redisMetaDataFactory =
-            gemFireCache.createInternalRegionFactory();
+      if ((redisData = cache.getRegion(REDIS_DATA_REGION)) == null) {
+        InternalRegionFactory<ByteArrayWrapper, RedisData> redisMetaDataFactory =
+            gemFireCache.createInternalRegionFactory(DEFAULT_REGION_TYPE);
         redisMetaDataFactory.addCacheListener(metaListener);
-        redisMetaDataFactory.setDataPolicy(DataPolicy.REPLICATE);
         redisMetaDataFactory.setInternalRegion(true).setIsUsedForMetaRegion(true);
-        redisMetaData = redisMetaDataFactory.create(REDIS_META_DATA_REGION);
+        redisData = redisMetaDataFactory.create(REDIS_DATA_REGION);
       }
 
-      keyRegistrar = new KeyRegistrar(redisMetaData);
+      keyRegistrar = new KeyRegistrar(redisData);
```

Comment:
This may be useful for reasons that I'm not immediately grasping, but do we still need the KeyRegistrar abstraction?

Reply:
Once all of our data types are converted over to just use the new DataRegion, then I think we will find that no code uses the KeyRegistrar methods anymore.